src/engine/enginemaster.cpp

```diff
@@ -97,6 +100,13 @@ EngineMaster::EngineMaster(UserSettingsPointer pConfig,
 
     m_pMasterDelay = new EngineDelay(group, ConfigKey(group, "delay"));
     m_pHeadDelay = new EngineDelay(group, ConfigKey(group, "headDelay"));
```

Comment:
Like latency compensation for mics, the scratching processing would not be delayed. The output from deck processing without timecode processing would be delayed, then timecode processing would be applied. This way, if the user scratches on beat, the scratch would be heard on beat.

Implementing this would complicate the application of the latency compensation delay in EngineMaster::process...